options/configurable.cc

```diff
@@ -53,13 +53,13 @@ Status Configurable::PrepareOptions(const ConfigOptions& opts) {
                 opt_info.AsRawPointer<Configurable>(opt_iter.opt_ptr);
             if (config != nullptr) {
               status = config->PrepareOptions(opts);
-              if (!status.ok()) {
-                return status;
-              }
             } else if (!opt_info.CanBeNull()) {
               status = Status::NotFound("Missing configurable object",
                                         map_iter.first);
             }
+            if (!status.ok()) {
+              return status;
+            }
```

Comment:
Do any of these early returns (either the removed one, or the new one) have any significance/is this a functional change? There's a `return status;` at the bottom of the method.

Reply:
The return was not deleted, it was moved.  In the original code, it would only return the last status as opposed to intermediary ones.